gammapy/datasets/map.py

```diff
@@ -1344,6 +1344,29 @@ def stat_sum(self):
         else:
             return cash_sum_cython(counts.ravel(), npred.ravel()) + prior_stat_sum
 
+    def stat_sum_asimov(self, model_name=None):
+        """Test statistic for the given model assuming an Asimov dataset
+
+        Parameters
+        ----------
+        model_name : str
+            Name of the sky model for which to compute the statistic
+        """
+
+        npred = self.npred().data
+        npred_source = self.npred_signal(model_names=[model_name]).data
+        npred_h0 = npred - npred_source
+
+        if self.mask is not None:
+            mask = self.mask.data
+            npred = npred[mask]
+            npred_h0 = npred_h0[mask]
+
+        L0 = cash_sum_cython(npred, npred)
+        L1 = cash_sum_cython(npred, npred_h0)
+
+        return np.sum(L1 - L0)
```

Comment:
I'm a little bit confused about the notation here 😅 
Let me sum this up for my own understanding:
1. You have a Map dataset built from an observation
2. `npred` is the number of the total number of signal and background counts predicted from all models
3. `npred_source` are the counts from one specific model
4. `npred_h0` is the null hypothesis, alias signal=0 and only bkg

So in total is `L0` the best fit profiled log-likelihood, or cash stat, and `L1` as the null hypo for only bkg. 
I have the following concerns:
- `npred_h0` being plugged into `L1` is a bit confusing, I would swap the names maybe
- this method is name `stat_sum_asimov` in reference to the normal `stat_sum` of the dataset although this is not a stat_sum but the test statistic itself for the null hypo, named $q_0$ in the Cowen paper
- This is not flexible enough and only computes the $q_0$ it is not possible to compute upper limits with this for example since one would need $q_\mu$ with $\mu$ being a strength parameter  that scales the signal model
- Priors are missing in this test statistic, they need be evaluated too and do not cancel since, see eq 26 cowen paper, the nuisance params are profiled for the null hypo and taken as given by the alternative hypo

Reply:
The last point actually includes reoptimizing the models for s=0